transport/peerlist.go

```diff
@@ -0,0 +1,15 @@
+package transport
+
+import "context"
+
+//go:generate mockgen -destination=transporttest/peerlist.go -package=transporttest go.uber.org/yarpc/transport PeerList
+
+// PeerList is a collection of Peers.  Outbounds request peers from the ChoosePeer to determine where to send requests
+type PeerList interface {
+	PeerSubscriber
```

Comment:
Offline discussion, decided to split the interfaces out, a PeerList will not implement the subscription so they don't need to be coupled.